Src/main.c

```diff
@@ -112,9 +117,17 @@ int main(void)
 
   /* USER CODE BEGIN 3 */
 
+  HAL_ADC_PollForConversion(&hadc, 20); //Laye - polls potentiometer for analog values 
+  converted_val = HAL_ADC_GetValue(&hadc); //Laye - return value of ADC 
+  compare_val = (converted_val/4096)*12000; // Laye - scaling 12 bit return value according to period of 12000
```

Comment:
The PWM should range from 1ms to 2ms of on-time each cycle, so you'll need to adjust the calculation to stay within that range.